docs/designs/data-models.md

```diff
@@ -0,0 +1,116 @@
+# Data Models Dictionary
+
+This document presents the data model used in this software. It describes different domain entities, fields within
+them, and relationships among the entities. Reading this document should provide a good overview of how the 
+product should work under-the-hood.
+
+Before you proceed, please read [terminologies.md](../terminologies.md) document to familiarize yourself with the 
+terms.
+
+## Entities
+This section describes the entites and fields within each of them.
+
+### User (aka Account)
```

Comment:
From the terminologies doc, it seems like a user can be affiliated with multiple Programs in different capacities like creator/collaborator, etc. Capturing this information, in the beginning, will help us add access restrictions in the future

Reply:
yes, will do